src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2804,13 +2811,31 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
 
 # help class handles runtime stack in format like 'node|6.1', 'php|5.5'
 class _StackRuntimeHelper:
-
-    def __init__(self, cmd, client, linux=False):
+    # pylint: disable=too-few-public-methods
+    class Runtime:
+        def __init__(self, display_name=None, configs=None, github_actions_properties=None, linux=False):
+            self.display_name = display_name
+            self.configs = configs if configs is not None else dict()
+            self.github_actions_properties = github_actions_properties
+            self.linux = linux
+
+    def __init__(self, cmd, linux=False, windows=False):
         self._cmd = cmd
-        self._client = client
+        self._client = web_client_factory(cmd.cli_ctx, api_version="2021-01-01")
         self._linux = linux
+        self._windows = windows
         self._stacks = []
 
+        # TODO try and get API support for this so it isn't hardcoded
```

Comment:
This was in the previous implementation of the runtimes handling (see [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L2896)). Windows runtimes are not set with a single attribute like linux runtimes (which uses `linuxFxVersion`), so we have to set them with the proper appsetting key and value. For function apps, the API tells you what appsettings to set, but the webapps API does not  

This is used only for `az webapp up` and `az webapp create`, so if the user deletes the appsetting after the creation of the webapp it will not impact either command.

As far as error handling, the appsetting is only set if the runtime has an entry in the config mapping, so there's no exception if a runtime doesn't have a value in the dict.